### PR TITLE
Remove Accept-Encoding header on redirect

### PR DESCRIPTION
**Pull request by mhashizume**
Prior to this commit, Puppet would copy all request headers in an HTTP redirect, including Accept-Encoding. In some cases when HTTP compression was enabled, the response would fail to get decompressed, then would fail to get parsed and trigger a vague error.

This commit strips the Accept-Encoding headers on redirect, allowing Ruby's built-in Net::HTTP to both compress and decompress the traffic.

Resolves https://github.com/puppetlabs/puppet/issues/9143

### CODE REVIEW

File: lib/puppet/http/redirector.rb
Comment (joshcooper) [inline lib/puppet/http/redirector.rb:61]: If `casecmp` returns `0`, then it's interpreted as truthy. I think you want:

```suggestion
      next if header.casecmp('Accept-Encoding').zero?
```

**Comment by mhashizume** [comment]
This will need to be backported to 7.x after merge.

**Review by AriaXLi — APPROVED** [review]

**Comment by mhashizume** [comment]
Blocked on https://github.com/puppetlabs/puppet/pull/9266

**Comment by joshcooper** [comment]
Needs rebase on main

**Pull request merged**
